src/core/overflowsafe_type.hpp

```diff
@@ -72,7 +78,12 @@ class OverflowSafeInt
 	OverflowSafeInt() : m_value(0) { }
 
 	OverflowSafeInt(const OverflowSafeInt& other) { this->m_value = other.m_value; }
-	OverflowSafeInt(const int64 int_)             { this->m_value = int_; }
+	OverflowSafeInt(const intmax_t int_)
+	{
+		this->m_value = unlikely(int_ > static_cast<intmax_t>(T_MAX)) ? T_MAX :
+				unlikely(int_ < static_cast<intmax_t>(T_MIN)) ? T_MIN :
+				static_cast<T>(int_);
+	}
```

Comment:
I would like to have a conversion constructor for `uintmax_t`; unfortunately, the compiler didn't seem to be able to resolve the ambiguity when offered the choice between `intmax_t` and `uintmax_t`.